Doc for electrodialysis flowsheet (#1385)
watertap-org/watertap: docs/technical_reference/flowsheets/electrodialysis_1stack_conc_recirc.rst

One-stack electrodialysis with a concentrate fluid recirculation
================================================================

Introduction
------------

> Electrodialysis (ED) is a promising technology for desalinating brackish waters and has been deployed at industrial scales [1]_. It utilizes electrical potential to drive ion diffusion through anion and cation exchange membranes, resulting in the dilution of the feed stream while producing a concentrated brine. A single ED stack is an assembly of multiple flow-by cells separated by alternating cation and anion exchange membranes positioned between a pair of electrodes. When a voltage is applied, ions in the cell pair are driven from one channel (forming the diluate channel) to the other (forming the concentrate channel), thereby desalinating water. A one-stack ED desalination system represents a basic ED operation, from which more complicated or larger-scale systems can be derived. Analyzing a one-stack ED system therefore provides valuable information on the technology's performance and cost-effectiveness for a given treatment task.

Reviewer: ```suggestion
Electrodialysis (ED) is a promising technology for desalinating brackish waters and has been deployed at industrial scales [1]_. It utilizes electrical potential to drive ion diffusion through anion and cation exchange membranes, resulting in the dilution of the feed stream while producing a concentrated brine. A single ED stack is an assembly of multiple flow-by cells separated by alternating cation and anion exchange membranes positioned between a pair of electrodes. When a voltage is applied, ions in the cell pair are driven from one channel (forming the diluate channel) to the other (forming the concentrate channel), thereby desalinating water. A one-stack ED desalination system represents a basic ED operation from which more complicated or larger-scale systems can be derived. Analyzing a one-stack ED system therefore provides valuable information on the technology's performance and cost-effectiveness for a given treatment task.
```

Equal flow conditions through the diluate and concentrate channels would result in a product water recovery of 50%. Larger water recoveries are commonly achieved by operating the system in a feed-and-bleed mode, where a portion of the concentrate outlet is recirculated back to its inlet, thus increasing the portion of product water from the diluate outlet. This flowsheet simulates a one-stack ED system operated in feed-and-bleed mode.

> 

Reviewer: Add a link here to the simpler One Stack ED documentation?

Implementation

> --------------

Reviewer: Should mention the objective function at the end of this section

> The modeled one-stack ED system is illustrated by Figure 1. The feed solution is split into two fluids through a separator unit, entering the diluate and concentrate channels of the ED stack. Two pump units are place respectively on the two channels entering pipelines to counterbalance the pressure drops across ED stack channels.  On the outlet side of the ED stack, all diluate fluids are collected into the total product water, and the concentrate fluids into the total brine stream.  A second separator unit takes a portion of the brine back into the concentrate pump inlet, with the rest being sent to brine disposal. Water recovery in this model is volume-based, i.e., the ratio of product volume to total volume of feed solution. The flowsheet relies on the following key assumptions:

Reviewer: ```suggestion
The modeled one-stack ED system is illustrated by Figure 1. The feed solution is split into two fluids through a separator unit, entering the diluate and concentrate channels of the ED stack. Two pump units are placed respectively on the two channels entering pipelines to counterbalance the pressure drops across ED stack channels.  On the outlet side of the ED stack, all diluate fluids are collected into the total product water, and the concentrate fluids into the total brine stream.  A second separator unit takes a portion of the brine back into the concentrate pump inlet, with the rest being sent to brine disposal. Water recovery in this model is volume-based, i.e., the ratio of product volume to total volume of feed solution. The flowsheet relies on the following key assumptions:
```

Documentation for unit models from WaterTAP: 
    * `Electrodialysis_1D <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/electrodialysis_1D.html>`_
    * `Pressure exchanger <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/pressure_exchanger.html>`_
Documentation for unit models from IDAES: 
    * `Feed block <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/feed.html>`_
    * `Separator <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/separator.html>`_
    * `Mixer <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/mixer.html>`_
    * `Product block <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/product.html>`_
Documentation for the property model:
    * `Multi-Component Aqueous Solution (MCAS) Property Package <https://watertap.readthedocs.io/en/latest/technical_reference/property_models/mc_aq_sol.html>`_

Degrees of Freedom
------------------

> The number of degree of freedom (DOF) is associated with the number of fixed variables (parameters) determined by the purpose of the modeling case. We implemented two modeling cases in the `main()` function: (1) the prediction of desalination outcome (salinity of the product water and saline disposal) and (2) the optimization of key decision variables in system design.  In the first case, DOF is set to zero by fixing all initial conditions of the feed solution fluid and definite ED stack parameters. All fixed values are presented in the section to follow.  In the second case, the values of those chosen to be the decision variables in the optimization are unfixed. The DOF number is therefore the number of decision variables. In this example, the decision variables are
> 

Reviewer: ```suggestion
The number of degrees of freedom (DOF) is associated with the number of fixed variables (parameters) determined by the purpose of the modeling case. We implemented two modeling cases in the flowsheet: (1) the prediction of desalination outcome (salinity of the product water and saline disposal) and (2) the optimization of key decision variables in system design.  In the first case, DOF is set to zero by fixing all initial conditions of the feed solution fluid and definite ED stack parameters. All fixed values are presented in the section to follow.  In the second case, the values of those chosen to be the decision variables in the optimization are unfixed. The DOF number is therefore the number of decision variables. In this example, the decision variables are

```

    * stack voltage applied
    * ED cell pair number 
    * ED cell length

> In the optimization example, the objective function is set to LCOW and the product water salinity is set to 0.1 :math:`g L^{-1}`.

Reviewer: Should include the LCOW equation and move it to the Implementation section

> One-stack electrodialysis with a concentrate fluid recirculation

Reviewer 2: Just for consistency with the Other "One-Stack Electrodialysis" and other fs titles
```suggestion
One-Stack Electrodialysis with a Concentrate Fluid Recirculation
```

Equal flow conditions through the diluate and concentrate channels would result in a product water recovery of 50%. Larger water recoveries are commonly achieved by operating the system in a feed-and-bleed mode, where a portion of the concentrate outlet is recirculated back to its inlet, thus increasing the portion of product water from the diluate outlet. This flowsheet simulates a one-stack ED system operated in feed-and-bleed mode.  A simpler ED flowsheet without fluid recirculation is presented as 

>      * `One-Stack Electrodialysis <https://watertap--1386.org.readthedocs.build/en/1386/technical_reference/flowsheets/electrodialysis_1stack.html>`_.

Reviewer 2: This link references the build from the PR instead of latest.
```suggestion
     * `One-Stack Electrodialysis <https://watertap.readthedocs.io/en/latest/technical_reference/flowsheets/electrodialysis_1stack.html>`_.
```

You can also check the discussion here between Marcus and I for other ways to link to other documentation. https://github.com/watertap-org/watertap/pull/1396/files/5ec96d06e3f6df729e71b9cb1e7c3a3a32da6015#diff-1009f2493fb3cb501cbd2ac5af69180ca4ba15b279a71519ed0fbe636de9dd14R14

 
 The product water salinity is set to 0.1 :math:`g L^{-1}`.

Documentation for unit models from WaterTAP: 
    * `Electrodialysis_1D <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/electrodialysis_1D.html>`_
    * `Pressure exchanger <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/pressure_exchanger.html>`_
Documentation for unit models from IDAES: 
    * `Feed block <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/feed.html>`_
    * `Separator <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/separator.html>`_
    * `Mixer <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/mixer.html>`_
    * `Product block <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/product.html>`_
Documentation for the property model:
    * `Multi-Component Aqueous Solution (MCAS) Property Package <https://watertap.readthedocs.io/en/latest/technical_reference/property_models/mc_aq_sol.html>`_

Degrees of Freedom
------------------

> The number of degrees of freedom (DOF) is associated with the number of fixed variables (parameters) determined by the purpose of the modeling case. We implemented two modeling cases in the flowsheet: (1) the prediction of desalination outcome (salinity of the product water and saline disposal) and (2) the optimization of key decision variables in system design.  In the first case, DOF is set to zero by fixing all initial conditions of the feed solution fluid and definite ED stack parameters. All fixed values are presented in the section to follow.  In the second case, the values of those chosen to be the decision variables in the optimization are unfixed. The DOF number is therefore the number of decision variables. In this example, the decision variables are

Reviewer 2: ```suggestion
The number of degrees of freedom (DOF) is associated with the number of fixed variables (parameters) determined by the purpose of the modeling case. There are two implemented modeling cases in the flowsheet: (1) the prediction of desalination outcome (salinity of the product water and saline disposal) and (2) the optimization of key decision variables in system design.  In the first case, DOF is set to zero by fixing all initial conditions of the feed solution fluid and definite ED stack parameters. All fixed values are presented in the section to follow.  In the second case, the values of those chosen to be the decision variables in the optimization are unfixed. The DOF number is therefore the number of decision variables. In this example, the decision variables are:
```

   "Cell pair number", ":math:`56`", ":math:`1`", "--"
   "Cell length", ":math:`1.68`", ":math:`m`", [2]_
   "Cell width", ":math:`0.197`", ":math:`m`",[5]_
   "Channel height", ":math:`7.1 \times 10^{-4}`", ":math:`m`", "--"
   "Water recovery", ":math:`70%`", ":math:`1`", "--"
   "Stack voltage", ":math:`10`", ":math:`V`", "--"
   "Thickness, aem and cem", ":math:`1.3 \times 10^{-5}`", ":math:`m`",[5]_
   "Areal resistance, aem", ":math:`1.77 \times 10^{-4}`", ":math:`\Omega m^2`", [5]_
   "Areal resistance, cem", ":math:`1.89 \times 10^{-4}`", ":math:`\Omega m^2`",[5]_
   "Water permeability, aem", ":math:`1.75 \times 10^{-14}`", ":math:`m s^{-1} Pa^{-1}`",[5]_
   "Water permeability, cem", ":math:`2.16 \times 10^{-14}`", ":math:`m s^{-1} Pa^{-1}`", [5]_
   "Water transport number, aem", ":math:`4.3`", ":math:`1`",[6]_
   "Water transport number, cem", ":math:`5.8`", ":math:`1`", [7]_
   "NaCl mass diffusivity, aem and cem", ":math:`3.28 \times 10^{-11}`", ":math:`m^2 s^{-1}`", [8]_
   "Spacer porosity", ":math:`0.83`", ":math:`1`", [2]_
   "Spacer specific surface area", ":math:`10400`", ":math:`m^{-1}`", [9]_
   "Pump efficiency", ":math:`0.8`", ":math:`1`", "--"

> References

Reviewer 2: Something is going on with these in the build that I'm not sure of
![image](https://github.com/watertap-org/watertap/assets/101219154/bc35e207-2f1e-4bdd-ba81-d42b2b0ec5f0)

Author: these are proper printout by the used referencing codes.  The numbers in "()" tell the subindices of places where that citation item is relevant.

One-stack electrodialysis with concentrate stream recirculation (feed-and-bleed mode)

> =====================================================================================

Reviewer: Is there a way to shorten this? Perhaps One-Stack Electrodialysis with Fluid Recirculation? Or maybe we just remove the "feed-and-bleed mode" from the title?
